rust/lancedb/src/query.rs

```diff
@@ -71,6 +73,7 @@ impl Query {
             use_index: true,
             filter: None,
             select: None,
+            select_with_transform: None,
```

Comment:
Given that only one of these can be set is it worth making this an enum?

```
enum Selection {
    All,
    Simple(Vec<String>),
    Projection(Vec<(String, String)>),
}
```

Though you'd have to move the error handling from `execute_stream` to the `select` / `select_with_transform` methods (e.g. raise an error in `select` if self.select != Select::All` and vice versa)